core/services/ccip/orm.go

```diff
@@ -79,82 +71,143 @@ func (o *orm) GetGasPricesByDestChain(ctx context.Context, destChainSelector uin
 func (o *orm) GetTokenPricesByDestChain(ctx context.Context, destChainSelector uint64) ([]TokenPrice, error) {
 	var tokenPrices []TokenPrice
 	stmt := `
-		SELECT DISTINCT ON (token_addr)
-		token_addr, token_price, created_at
+		SELECT token_addr, token_price
 		FROM ccip.observed_token_prices
-		WHERE chain_selector = $1
-		ORDER BY token_addr, created_at DESC;
+		WHERE chain_selector = $1;
 	`
 	err := o.ds.SelectContext(ctx, &tokenPrices, stmt, destChainSelector)
 	if err != nil {
 		return nil, err
 	}
-
 	return tokenPrices, nil
 }
 
-func (o *orm) InsertGasPricesForDestChain(ctx context.Context, destChainSelector uint64, jobId int32, gasPrices []GasPriceUpdate) error {
+func (o *orm) UpsertGasPricesForDestChain(ctx context.Context, destChainSelector uint64, gasPrices []GasPrice) (int64, error) {
 	if len(gasPrices) == 0 {
-		return nil
+		return 0, nil
+	}
+
+	uniqueGasUpdates := make(map[string]GasPrice)
+	for _, gasPrice := range gasPrices {
+		key := fmt.Sprintf("%d-%d", gasPrice.SourceChainSelector, destChainSelector)
+		uniqueGasUpdates[key] = gasPrice
 	}
 
-	insertData := make([]map[string]interface{}, 0, len(gasPrices))
-	for _, price := range gasPrices {
+	insertData := make([]map[string]interface{}, 0, len(uniqueGasUpdates))
+	for _, price := range uniqueGasUpdates {
 		insertData = append(insertData, map[string]interface{}{
 			"chain_selector":        destChainSelector,
-			"job_id":                jobId,
 			"source_chain_selector": price.SourceChainSelector,
 			"gas_price":             price.GasPrice,
 		})
 	}
 
-	// using statement_timestamp() to make testing easier
-	stmt := `INSERT INTO ccip.observed_gas_prices (chain_selector, job_id, source_chain_selector, gas_price, created_at)
-		VALUES (:chain_selector, :job_id, :source_chain_selector, :gas_price, statement_timestamp());`
-	_, err := o.ds.NamedExecContext(ctx, stmt, insertData)
+	stmt := `INSERT INTO ccip.observed_gas_prices (chain_selector, source_chain_selector, gas_price, created_at)
+		VALUES (:chain_selector, :source_chain_selector, :gas_price, statement_timestamp())
+		ON CONFLICT (source_chain_selector, chain_selector)
+		DO UPDATE SET gas_price = EXCLUDED.gas_price, created_at = EXCLUDED.created_at;`
+
+	result, err := o.ds.NamedExecContext(ctx, stmt, insertData)
 	if err != nil {
-		err = fmt.Errorf("error inserting gas prices for job %d: %w", jobId, err)
+		return 0, fmt.Errorf("error inserting gas prices %w", err)
 	}
-
-	return err
+	return result.RowsAffected()
```

Comment:
RowsAffected is returned and ignored, is there any case where we should actually check it and act accordingly, maybe logging only?

Reply:
Yeah, it's only for monitoring purposes. Client of the API won't be interested in that information in most of the cases. However, it's a valid info to understand how many rows and how often we update. Based on that info we can tune the query or tune update intervals ;)